### PR TITLE
The Brickening

### PR DESCRIPTION
**Pull request by GuppyLaxx**
Adds bricks, a decent enough melee weapon, and great thrown weapon, SPECIALLY to break into places.
They spawn in maintenance, and can even be bought by traitors 2 TC a pop.
Hoping to make the sprites better and add some meme sounds by tomorrow but this shit is good like it is if you wanna add it already.
:cl: GuppyLaxx
add: Added bricks. Break the window and hope the alarm doesn't go off.
/:cl:

### CODE REVIEW

File: hippiestation/code/modules/uplink/uplink_items.dm [hunk 1]
```diff
@@ -340,3 +340,10 @@
 	item = /obj/item/storage/box/syndie_kit/synth
 	cost = 8
 	surplus = 40
+
+/datum/uplink_item/badass/brick
+	name = "A brick"
+	desc = "A literal brick, able to break a common windows like nothing. Servers well as a melee and thrown weapon aswell. Keep an eye out for the fabled brown brick."
```
Comment (yoyobatty) [inline hippiestation/code/modules/uplink/uplink_items.dm]: Serves*

**Comment by Manly-Dwarf** [comment]
brown bricks

**Comment by TrickyDickRichard** [comment]
in minecraft

File: hippiestation/code/game/objects/items/weaponry.dm [hunk 2]
```diff
@@ -147,3 +147,54 @@
...
+	force = 12 // decent weapon
+	throwforce = 15 // good throw
+	attack_verb = list("bricked")
+	var/durability = 5
+
+/obj/item/brick/Initialize()
+	..()
+	if(prob(0.5))
+		name = "brown brick"
+		desc = "<font color = #835C3B>I understand why all the kids are playing this game these days. It's because they like to build brown bricks with Minecrap. I also like to build brown bricks with Minecrap. It's the most fun you can possibly have.</font color>"
```
Comment (JohnGinnane) [inline hippiestation/code/game/objects/items/weaponry.dm]: Remove "color" from the `</font color>`

File: hippiestation/code/game/objects/items/weaponry.dm [hunk 3]
```diff
@@ -147,3 +147,54 @@
...
+			if(prob(1) && M.stat != DEAD)
+				M.emote("scream")
+				M.visible_message("<span class='danger'>[user] knocks out [M] with [src]!</span>", \
+								"<span class='userdanger'>[user] knocks out [M] with [src]!</span>")
+				M.SetSleeping(60)
+				M.adjustBrainLoss(5)
+	..()
+
+/obj/item/brick/throw_impact(atom/hit_atom)
+	if(!..())
```
Comment (JohnGinnane) [inline hippiestation/code/game/objects/items/weaponry.dm]: Why are you checking for `!..()`

Comment (GuppyLaxx) [reply hippiestation/code/game/objects/items/weaponry.dm]: oh i think i left this over from way back, iirc it checks if you caught it or something but i probably fucked up

Comment (JohnGinnane) [reply hippiestation/code/game/objects/items/weaponry.dm]: No you're right, it must be for catching it, looking at other code

Comment (JohnGinnane) [reply hippiestation/code/game/objects/items/weaponry.dm]: image

**Comment by earthmoonjune** [comment]
+1

File: hippiestation/code/game/objects/items/weaponry.dm [hunk 4]
```diff
@@ -147,3 +147,56 @@
...
+
+/obj/item/brick/attack(mob/living/target, mob/living/user)
+	if(ishuman(target))
+		var/mob/living/carbon/human/M = target
+		if(!istype(M.head, /obj/item/clothing/head/helmet) && user.zone_selected == BODY_ZONE_HEAD)
+			if(prob(1) && M.stat != DEAD)
+				M.emote("scream")
+				M.visible_message("<span class='danger'>[user] knocks out [M] with [src]!</span>", \
+								"<span class='userdanger'>[user] knocks out [M] with [src]!</span>")
+				M.SetSleeping(60)
```
Comment (Jujumatic) [inline hippiestation/code/game/objects/items/weaponry.dm]: ouch lol

**Review by Jujumatic — DISMISSED** [review]
looks gucci

**Comment by ghost** [comment]
Please get better sprites.
After that, it'll have my +1

File: hippiestation/code/game/objects/items/weaponry.dm (hunk 3 above)
Comment (ghost) [reply hippiestation/code/game/objects/items/weaponry.dm]: do the one john posted since this won't return the value ..() at the end of the proc

**Review by ghost — DISMISSED** [review]
does durability even do anything in this code?
you define it as a new variable in this object and all you do is deduct from it whenever it hits something

**Comment by GuppyLaxx** [comment]
yes, it's checked whenever it's thrown at a window/person, so you can't spam bricks at windows 24/7 to break them, nor throw spam the shit out of one brick at people and knock them out

File: hippiestation/code/game/objects/items/weaponry.dm (hunk 4 above)
Comment (Carbonhell) [reply hippiestation/code/game/objects/items/weaponry.dm]: Don't think this is the right proc to call, shouldn't it be knockdown or weaken

Comment (GuppyLaxx) [reply hippiestation/code/game/objects/items/weaponry.dm]: It knocks someone unconscious. Not stun them down to the floor.

File: hippiestation/code/game/objects/items/weaponry.dm [hunk 5]
```diff
@@ -147,3 +147,57 @@
...
+	. = ..()
+	if(!.)
+		if(istype(hit_atom, /obj/structure/window) && durability)
+			var/obj/structure/window/W = hit_atom
+			W.take_damage(throwforce*10, BRUTE, "melee", 0)
+			durability -= 1
+
+		if(ishuman(hit_atom))
+			var/mob/living/carbon/human/H = hit_atom
+			if(prob(10) && !istype(H.head, /obj/item/clothing/head/helmet) && durability) // I couldnt figure out how to make it check if it's hitting the head so I just made it check for helmet, sry bby
```
Comment (Carbonhell) [inline hippiestation/code/game/objects/items/weaponry.dm:196]: Ew,needs to be fixed

Comment (GuppyLaxx) [reply hippiestation/code/game/objects/items/weaponry.dm:196]: unless you want me to make it check if the person throwing is aiming for head there's no fix for checking what limb it hit

File: hippiestation/code/game/objects/items/weaponry.dm (hunk 4 above)
Comment (Manly-Dwarf) [reply hippiestation/code/game/objects/items/weaponry.dm]: No, no. That's wrong. Use AdjustUnconscious.

File: hippiestation/code/game/objects/items/weaponry.dm (hunk 5 above)
Comment (Manly-Dwarf) [reply hippiestation/code/game/objects/items/weaponry.dm:196]: Ehh that might just be the way to go about it, honestly. Check for if they're throwing at the head, and then check for helmet. Maybe add some bonus to throwing it at the chest or other limbs so knockout isn't the only way to go.

Comment (GuppyLaxx) [reply hippiestation/code/game/objects/items/weaponry.dm:196]: checking if they're aiming for head isn't really gonna change anything

File: hippiestation/code/game/objects/items/weaponry.dm [hunk 6]
```diff
@@ -147,3 +147,57 @@
...
+	throwforce = 15 // good throw
+	attack_verb = list("bricked")
+	hitsound = 'hippiestation/sound/effects/brick.ogg'
+	var/durability = 5
+
+/obj/item/brick/Initialize()
+	..()
+	if(prob(0.5))
+		name = "brown brick"
+		desc = "<font color = #835C3B>I understand why all the kids are playing this game these days. It's because they like to build brown bricks with Minecrap. I also like to build brown bricks with Minecrap. It's the most fun you can possibly have.</font>"
```
Comment (Kawaii-Big-Boss) [inline hippiestation/code/game/objects/items/weaponry.dm]: most fun you can possibly have in an app*

**Review by Carbonhell — DISMISSED** [review]

File: hippiestation/code/game/objects/items/weaponry.dm [hunk 7]
```diff
@@ -147,3 +147,57 @@
...
+	if(ishuman(target))
+		var/mob/living/carbon/human/M = target
+		if(!istype(M.head, /obj/item/clothing/head/helmet) && user.zone_selected == BODY_ZONE_HEAD)
+			if(prob(1) && M.stat != DEAD)
+				M.emote("scream")
+				M.visible_message("<span class='danger'>[user] knocks out [M] with [src]!</span>", \
+								"<span class='userdanger'>[user] knocks out [M] with [src]!</span>")
+				M.AdjustUnconscious(60)
+				M.adjustBrainLoss(5)
+	..()
```
Comment (Carbonhell) [inline hippiestation/code/game/objects/items/weaponry.dm]: shouldn't this be called before, actually?

Comment (GuppyLaxx) [reply hippiestation/code/game/objects/items/weaponry.dm]: I just did it so it takes priority/comes first

**Review by Carbonhell — APPROVED** [review]

**Pull request merged**